### PR TITLE
Download protoc instead of using embedded version

### PR DESCRIPTION
**Pull request by robodenitro**
### Download protoc binary
This PR removes preliminary downloaded protoc binaries. Now protoc wrapper downloads protoc from upstream if the version doesn't exists in local cache.

### Other changes
- Update protoc to v3.21.2
- Update go-git to v5
- Refresh Google proto files

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -334,6 +382,40 @@ func runProtoc() int {
...
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("bad status: %s", resp.Status)
+	}
+
+	_, err = io.Copy(out, resp.Body)
+	if err != nil {
+		return err
```
Comment (mier85) [inline main.go:428]: if the download failed for some reason, should we delete the partial file? will the application be able to recover without deleting the cache directory manually?

File: main.go [hunk 2]
```diff
@@ -282,21 +294,57 @@ func cacheFile(path ...string) string {
...
-		// Checksum mismatch, create a new binary in the temporary directory
-		err = ioutil.WriteFile(protocExePath, protoc, 0755)
+
+	if _, err := os.Stat(protocExePath); err == nil {
+		return protocExePath, nil
+	}
+
+	log.Println("saving protoc to path: ", protocExePath)
+
+	url := fmt.Sprintf("%[1]s/%[2]s/protoc-%[2]s-%[3]s.exe", protoBinariesBaseURL, version, arch)
```
Comment (ynori7) [inline main.go:328]: Is it always a `.exe` even for Linux?

Comment (robodenitro) [reply main.go:328]: yes it's been always defined this way

File: main.go (hunk 1 above)
Comment (robodenitro) [reply main.go:428]: I added handler for this situation to downloadProtoc() , it's in defer statement

**Review by mier85 — APPROVED** [review]

**Pull request merged**